src/App.js

```diff
@@ -2,56 +2,51 @@ import React from 'react';
 import Header from './components/Header.js';
 import images from './components/data.js';
 import ImageList from './components/ImageList.js';
+import Dropdown from './components/Dropdown.js';
 import './App.css';
 
 export default class App extends React.Component {
   //State has a keyword that can start as an empty string-> no images, don't use the array use keyword or horns from data.js 
   state = {
-    keyword: ''
+    keyword: '',
+    horns: '',
     }
 
-    handleChange = (e) => {
+    handleKeyword = (e) => {
       this.setState({ keyword: e.target.value})
     }
+    
+    handleHorns = (e) => {
+      this.setState({horns: Number(e.target.value)})
+    }
   
   //function to filter the images array with the state option. Line 9.5 filter array = function(images). return all images that match the keyword in State
   render() {
-    const filteredImages = images.filter((image) => {
+    const filteredImages = images
+    .filter((image) => {
       if (this.state.keyword === '') return true
       if (image.keyword === this.state.keyword) return true
       return false
      }) 
-    return (
-      <>
+     .filter((image) => {
+      if (this.state.horns === '') return true
+      if (image.horns === this.state.horns) return true
+      return false
+     })
+
+     //get an array for just the keywords and get an array for all the horn options (Diyana helped here)
+     const keywordsArray = images.map(entry=>entry.keyword)
```

Comment:
This method currently makes an array with lots of repeats, to avoid repeats:
`let keywordsArray = [];
images.map(entry => {
if(!keywordsArray.entry.keyword) keywordsArray.push(entry.keyword)
})`
This allows you to make sure there are no repeats by initializing an empty array, and then when it maps it looks at the keywordArray to make sure the word does not already exist inside it, if it does not it pushes it into the array, if it does it just skips it.